Xamarin.Forms.Xaml/ApplyPropertiesVisitor.cs

```diff
@@ -290,148 +267,204 @@ static string GetContentPropertyName(IEnumerable<CustomAttributeData> attributes
+				return;
 
-				((BindableObject)xamlelement).SetBinding(property, value as BindingBase);
+			//If it's an already initialized property, add to it
+			if (xpe == null && TryAddToProperty(xamlelement, localName, value, lineInfo, serviceProvider, out xpe))
 				return;
+
+			xpe = xpe ?? new XamlParseException($"Cannot assign property \"{localName}\": Property does not exists, or is not assignable, or mismatching type between value and property", lineInfo);
+			if (context.DoNotThrowOnExceptions)
+				System.Diagnostics.Debug.WriteLine(xpe.Message);
+			else
+				throw xpe;
+		}
+
+		static bool TryConnectEvent(object element, string localName, object value, object rootElement, IXmlLineInfo lineInfo, out Exception exception)
+		{
+			exception = null;
+
+			var elementType = element.GetType();
+			var eventInfo = elementType.GetRuntimeEvent(localName);
+			var stringValue = value as string;
+
+			if (eventInfo == null || IsNullOrEmpty(stringValue))
+				return false;
+
+			var methodInfo = rootElement.GetType().GetRuntimeMethods().FirstOrDefault(mi => mi.Name == (string)value);
+			if (methodInfo == null) {
+				exception = new XamlParseException($"No method {value} found on type {rootElement.GetType()}", lineInfo);
+				return false;
 			}
 
-			//If it's a BindableProberty, SetValue
+			try {
+				eventInfo.AddEventHandler(element, methodInfo.CreateDelegate(eventInfo.EventHandlerType, rootElement));
+				return true;
+			} catch (ArgumentException ae) {
+				exception = new XamlParseException($"Method {stringValue} does not have the correct signature", lineInfo, ae);
+			}
+			return false;
+		}
+
+		static bool TrySetDynamicResource(object element, BindableProperty property, object value, IXmlLineInfo lineInfo, out Exception exception)
+		{
+			exception = null;
+
+			var elementType = element.GetType();
+			var dynamicResource = value as DynamicResource;
+			var bindable = element as BindableObject;
+
+			if (dynamicResource == null || property == null)
+				return false;
+
+			if (bindable == null) {
+				exception = new XamlParseException($"{elementType.Name} is not a BindableObject", lineInfo);
+				return false;
+			}
+
+			bindable.SetDynamicResource(property, dynamicResource.Key);
+			return true;
+		}
+
+		static bool TrySetBinding(object element, BindableProperty property, string localName, object value, IXmlLineInfo lineInfo, out Exception exception)
+		{
+			exception = null;
+
+			var elementType = element.GetType();
+			var binding = value as BindingBase;
+			var bindable = element as BindableObject;
+			var nativeBindingService = DependencyService.Get<INativeBindingService>();
+
+			if (binding == null)
+				return false;
+
+			if (bindable != null && property != null) {
+				bindable.SetBinding(property, binding);
+				return true;
+			}
+
+			if (nativeBindingService != null && property != null && nativeBindingService.TrySetBinding(element, property, binding))
```

Comment:
Warning, new stuff:

try to set the binding via the native binding service, which will then turn that into a native binding